src/frontend/src/binder/values.rs

```diff
@@ -135,10 +122,11 @@ impl Binder {
         };
 
         let values_id = self.next_values_id();
+        #[expect(clippy::disallowed_methods)]
         let schema = Schema::new(
             types
                 .into_iter()
-                .zip_eq_fast(0..num_columns)
+                .zip(0..num_columns)
```

Comment:
I suggest we first take a prefix of the `expected_types` before passing into this function, so their lengths can be always the same.